### PR TITLE
Drop unused `device_max_stream_id` table

### LINKED ISSUE #7887: drop the `device_max_stream_id` table
it's unused since #7882

### PR DESCRIPTION
**Pull request by vgerak**
Closes #7887. 

Signed-off-by: Vasilis Gerakaris <vasilis.gerakaris@navarino.gr>

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)
* [x] Code style is correct (run the [linters](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#code-style))

### CODE REVIEW

File: synapse/storage/databases/main/schema/delta/59/drop_device_max_stream_id.sql [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/* Copyright 2020 The Matrix.org Foundation C.I.C
```
Comment (richvdh) [inline synapse/storage/databases/main/schema/delta/59/drop_device_max_stream_id.sql:1]: firstly, this file is in the wrong directory: the current schema version (as per `prepare_database.py`) is 58. (I see we have a file that has crept into the wrong directory by mistake: I'll follow up on that elsewhere.)

Secondly, delta files should be numbered to make sure they run in the right order. Please could you rename it to `21drop_device_max_stream_id.sql` to fit in https://github.com/matrix-org/synapse/tree/develop/synapse/storage/databases/main/schema/delta/58?

Thirdly, as you don't work for the matrix.org foundation, you can't write this copyright line. You can either put your own name there or omit the copyright line altogether.

**Review by richvdh — CHANGES_REQUESTED** [review]
thanks very much for this! Unfortunately a couple of cleanup tasks are needed...

File: synapse/storage/databases/main/schema/delta/59/drop_device_max_stream_id.sql (hunk 1 above)
Comment (vgerak) [reply synapse/storage/databases/main/schema/delta/59/drop_device_max_stream_id.sql]: Thanks for the notes @richvdh !

File moved to directory (I found the folder and assumed that {version+1} would be the migrations for the next version so I originally put it there.

Regarding the copyright, I thought that with the notice I would be "surrendering" the copyright to the Matrix.org foundation; apparently that was not the case.

File renamed & moved under version 58, and copyright notice removed, as requested.

**Review by richvdh — APPROVED** [review]
lgtm, thank you!

**Pull request merged**